### ISSUE TITLE
GUI not working

### ISSUE THREAD

**Issue by xbotuk** [microsoft/wslg#820]
### Windows build number:

22000.708

### Your Distribution version:

22.04

### Your WSL versions:

WSL version: 0.65.3.0
Kernel version: 5.15.57.1
WSLg version: 1.0.41
MSRDC version: 1.2.3213
Direct3D version: 1.601.0
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22000.708

### Steps to reproduce:

run xeyes or any other GUI program it produces 

user@mypc:~$ xeyes
Error: Can't open display: :0

### WSL logs:

[stderr.log](https://github.com/microsoft/wslg/files/9438657/stderr.log)
[weston.log](https://github.com/microsoft/wslg/files/9438658/weston.log)
[pulseaudio.log](https://github.com/microsoft/wslg/files/9438659/pulseaudio.log)

### WSL dumps:

no files in /mnt/wslg/dumps

### Expected behavior:

the UI program should start but it doesn't

### Actual behavior:

Error: Can't open display: :0

**Comment by carylewis** [comment]
I believe that this is the same problem that I am seeing.

wslg version 0.47 does not manifest this problem.

What appears to be happening is that the newer wslg versions somehow link their /mnt/wlsg/.X11-unix directory to /tmp/.X11-unix in the distribution, in my case Ubuntu.

During the ubuntu initialization, the systemd startup services remove everything in /tmp. This is now also removing the X0 file in /mnt/wslg/.X11-unix directory. So the old posts about doing a ln -s no longer work.

I worked around this issue by modifying the `/usr/lib/tmpfiles.d/x11.conf` configuration file, and commenting out all the lines:

```
#  This file is part of systemd.
#
#  systemd is free software; you can redistribute it and/or modify it
#  under the terms of the GNU Lesser General Public License as published by
#  the Free Software Foundation; either version 2.1 of the License, or
#  (at your option) any later version.

# See tmpfiles.d(5) for details

# Make sure these are created by default so that nobody else can
# or empty them at startup
#D! /tmp/.X11-unix 1777 root root 10d
#D! /tmp/.ICE-unix 1777 root root 10d
#D! /tmp/.XIM-unix 1777 root root 10d
#D! /tmp/.font-unix 1777 root root 10d
#D! /tmp/.Test-unix 1777 root root 10d

# Unlink the X11 lock files
#r! /tmp/.X[0-9]*-lock
```
I hope this helps.

Btw, I tried disabling the systemd-tmpfiles-setup service, but that didn't work.

If anyone can provide details as to how the later versions of wslg export their /mnt/wslg/.X11-unix directory to the distributions, that would be most appreciated.

**Comment by rye2020** [comment]
did not work for me.

**Comment by armleo** [comment]
Non of workaround worked for me. Completly removed Ubuntu, WSL, Virtual machines, etc from optional features then reinstalled again

**Comment by crisperit** [comment]
@carylewis 

Thanks - you've saved me!!

**Comment by tusharsnn** [comment]
I found that sometimes wslg creates `/tmp/.X11-unix/X1` instead of `/tmp/.X11-unix/X0` but still exports `DISPLAY=:0` which will not work as apps will try to find **X0**, in this case exporting `DISPLAY=:1` should make it work. note that if wslg creates `/tmp/.X11-unix/X<num>` then you need to export `DISPLAY=:<num>` respectively.

**Comment by hideyukn88** [comment]
Thanks all for reporting the issue and analyzing, I think we understand the issue, but would like to double check, do you install and run multiple Linux distros on WSL when this issue occurs? thanks!

**Comment by tusharsnn** [comment]
I am not using multiple distros, but this issue is most notable when I close wsl with `wsl --shutdown` and quickly open another terminal tab. While going through the weston source I also found that it does create X1 if X0 already exists but cannot verify that, as by the time terminal opens its gone.

**Comment by hideyukn88** [comment]
@tusharsnn, your observation of weston code is right, and there is recent change in sharing instance of system-distro, and this leads multiple distro to share same instance of system distro file system accidentally, and that causes second distro to see existing X0 lock file by first distro, then end up X11-unix on incremented index. Since the timing of the raising the issue report on this problem and that WSL change introduces is closely aligned, so we believe the reverting back system-distro sharing would solve the issue, thanks!

**Comment by hideyukn88** [comment]
This issue is fixed by WSL version [0.67.6](https://github.com/microsoft/WSL/releases/tag/0.67.6) or newer.

**Comment by hideyukn88** [comment]
Hi all, WSL 0.70.0 is just released from Store, https:://aka.ms/wslstorepage, and this should address the issue, please update it by `wsl --update`, and if the problem still persists, please let us know by reopening this issue, thanks!
